pkg/polaris/graphql/sla/sla_domain_list.go

```diff
@@ -97,6 +98,32 @@ type Domain struct {
 	Version           string            `json:"version"`
 }
 
+// DomainByID returns the global SLA domain with the specified ID.
+func DomainByID(ctx context.Context, gql *graphql.Client, id uuid.UUID) (Domain, error) {
+	gql.Log().Print(log.Trace)
+	query := slaDomainQuery
+	buf, err := gql.Request(ctx, query, struct {
+		ID uuid.UUID `json:"slaDomainId"`
+	}{ID: id})
+	if err != nil {
+		return Domain{}, graphql.RequestError(query, err)
+	}
+
+	var payload struct {
+		Data struct {
+			Result Domain `json:"result"`
+		} `json:"data"`
+	}
+	if err := json.Unmarshal(buf, &payload); err != nil {
+		return Domain{}, graphql.UnmarshalError(query, err)
+	}
+	if payload.Data.Result.Archived {
+		return Domain{}, graphql.ErrNotFound
```

Comment:
Could we make this error a little bit more descriptive, in case it's printed out by Terraform. E.g:
```
return Domain{}, fmt.Errorf("sla domain %q: %w", id, graphql.ErrNotFound)
```